card/mod/01_core/set/all/templating.rb

```diff
@@ -12,29 +12,34 @@ def template
   # note that a *default template is never returned for an existing card.
   @template ||= begin
     @virtual = false
+
+    # NEW CARDS
     if new_card?
-      default_card = rule_card :default, :skip_modules=>true
+      default = rule_card :default, :skip_modules=>true
 
       dup_card = dup
-      dup_card.type_id = default_card ? default_card.type_id : Card.default_type_id
-
+      dup_card.type_id = default ? default.type_id : Card.default_type_id
 
-      if content_card = dup_card.structure_rule_card
+      if structure = dup_card.structure_rule_card
         @virtual = true if junction?
-        content_card
+        self.type_id = structure.type_id if assign_type_to?(structure)
```

Comment:
Sorry about the imprecise language again, but you understood.  It just feels to me like it is still pretty virtual when the structure rule is over a real card, but that is the definition we use for coding reasons like this.

Makes sense, for this case we would never do validation, so the type never gets set on the actual card (in memory, since by definition it doesn't get saved).  The case of a virtual card getting saved always worked because it would happen before validation sometime or maybe even in validation.